src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -226,6 +226,10 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 		// Add connection profile to parent group and update group children. Then reveal and expand the new connection
 		this._register(this._connectionManagementService.onConnectionProfileCreated(async (newConnection) => {
 			if (this._tree instanceof AsyncServerTree) {
+				// If a connection is added to an empty tree, there won't be any input set. So set the input here
+				if (!this._tree.getInput()) {
+					this._tree.setInput(TreeUpdateUtils.getTreeInput(this._connectionManagementService));
```

Comment:
This seems like it's supposed to be done as part of `handleOnCapabilitiesRegistered` - is the issue here that the user is trying to add a connection before the capabilities are registered?

Is there any reason we wouldn't just initialize the tree with a call to this when it's first created so we're always sure to have something? Otherwise we could easily miss a code path that tries to add/modify the tree in some other way.

Reply:
The default group does not exist until the connection is added in ads. The group gets created only when we add a connection in ads.

Reply:
Sigh, what a great design....

Ok - could you at least add that context here for future reference?